clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -308,7 +308,7 @@ protected Record readNext(long baseOffset, long baseTimestamp, int baseSequence,
                 try {
                     return DefaultRecord.readFrom(buffer, baseOffset, baseTimestamp, baseSequence, logAppendTime);
                 } catch (BufferUnderflowException e) {
-                    throw new InvalidRecordException("Incorrect declared batch size, premature EOF reached");
+                    throw new InvalidRecordException("Incorrect declared batch size, premature EOF reached", e);
```

Comment:
Does the underflow exception actually add any information? If I recall correctly, it does not.

Reply:
You are right, it doesn't. I will revert this.